publisher/release/marker.go

```diff
@@ -0,0 +1,55 @@
+package release
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// Mark represents a release mark. It will contain the name of the release (appName, tag...)
+// and the start and end of a release
+// When the release has been started, the end will be zero
+type Mark struct {
+	AppName   string     `json:"app_name"`
+	Tag       string     `json:"tag"`
+	RunID     string     `json:"run_id"`
+	Start     CustomTime `json:"start"`
+	End       CustomTime `json:"end"`
+	RepoName  string     `json:"repo_name"`
+	Schema    string     `json:"schema"`
+	SchemaURl string     `json:"schema_url"`
+}
+
+// Marker abstracts the persistence of the start and end of a release
+type Marker interface {
+	Start(appName string, tag string, runID string, repoName string, schema string, schemaURL string) (Mark, error)
```

Comment:
Nit: considering all these arguments I wonder if it would make sense including an additional struct. Something like this:

```go
type RealeseInfo struct {
	AppName   string     `json:"app_name"`
	Tag       string     `json:"tag"`
	RunID     string     `json:"run_id"`
	RepoName  string     `json:"repo_name"`
	Schema    string     `json:"schema"`
	SchemaURl string     `json:"schema_url"`
}

type Mark struct {
    ReleaseInfo
	Start     CustomTime `json:"start"`
	End       CustomTime `json:"end"`
}

type Marker interface {
	Start(releaseInfo ReleaseInfo) (Mark, error)
	End(mark Mark) error
}
```

WDYT?

Reply:
I like it. I thought about using `Mark` but I didn't like setting Start from outside. I'll add this struct, thanks!

Reply:
Added it in 5dc1f5315e76a306e74bce9d3055f05eec1bc7f6